lib/core/kotekanTrackers.hpp

```diff
@@ -0,0 +1,110 @@
+#ifndef KOTEKAN_TRACKERS_HPP
+#define KOTEKAN_TRACKERS_HPP
+
+#include "restServer.hpp" // for connectionInstance, restServer
+#include "visUtil.hpp"    // for StatTracker
+
+#include <map>      // for map
+#include <memory>   // for shared_ptr
+#include <mutex>    // for mutex
+#include <stddef.h> // for size_t
+#include <string>   // for string
+
+namespace kotekan {
+
+typedef std::map<std::string, std::shared_ptr<StatTracker>> stage_trackers_t;
+
+class KotekanTrackers {
+
+public:
+    /**
+     * @brief Returns an instance of the kotekan trackers.
+     *
+     * @return Returns the kotekan trackers instance.
+     */
+    static KotekanTrackers& instance();
+
+    /**
+     * @brief Registers this class with the REST server, creating the
+     *        /trackers end point
+     * @param rest_server The server to register with.
+     */
+    void register_with_server(restServer* rest_server);
+
+    /**
+     * @brief The call back function for the REST server to use.
+     * This returns all contents from every tracker and grouped by stage.
+     *
+     * This function is never called directly.
+     *
+     * @param conn The connection instance to send results to.
+     */
+    void trackers_callback(connectionInstance& conn);
+
+    /**
+     * @brief The call back function for the REST server to use.
+     * This returns min/max/avg/std of each tracker and grouped by stage.
+     *
+     * This function is never called directly.
+     *
+     * @param conn The connection instance to send results to.
+     */
+    void trackers_current_callback(connectionInstance& conn);
+
+    /**
+     * @brief Adds a new tracker
+     *
+     * @param stage_name The name of the stage.
+     * @param tracker_name The name of the tracker.
+     * @param unit The unit of the tracker.
+     * @param size The size of the tracker with default of 100.
+     * @param is_optimized The switch of min/max methods of the tracker.
```

Comment:
Maybe add something like "See `statTracker` for details."  Since otherwise it might be unclear where to get information on this flag.